internal/beater/java_attacher/java_attacher.go

```diff
@@ -167,6 +177,71 @@ func (j *JavaAttacher) Run(ctx context.Context) error {
 	}
 }
 
+func (j *JavaAttacher) deleteTempDir(tempDir string) {
+	func(name string) {
+		err := os.RemoveAll(name)
+		if err != nil {
+			j.logger.Errorf("failed to delete Java attacher temp dir at %v: %v", tempDir, err)
+		}
+	}(tempDir)
+}
+
+// Creates a temp dir for the java attacher jar with the attacher jar in it, both with 0755 access permissions.
+// The assumption is that the default temp dir will be readable for all users.
+// However, there may be systems in which the temp directory is user-specific,
+// in which case we'll need to modify this function to run a temp dir search per user,
+// maybe by externally executing `echo XXX` and reading the output, relying on environment
+// variables as implemented in os.TempDir, or looking in known paths.
+// In such case, we would switch to using a cache of temp-agent-dir per user.
+func (j *JavaAttacher) createAttacherTempDir() (string, error) {
+	tempDir, err := os.MkdirTemp("", "elasticapmagent")
+	if err != nil {
+		return "", err
+	}
+	// todo: check if we need to use different value for specific OSs
+	// make sure this dir is readable and executable by all users and writable only by the current user
+	err = os.Chmod(tempDir, 0755)
+	if err != nil {
+		j.logger.Errorf("failed to change permissions for %v to make it readable for all users: %v", tempDir, err)
+	}
+	bundledAttacherFile, err := os.Open(bundledJavaAttacher)
+	if err != nil {
+		defer j.deleteTempDir(tempDir)
+		return "", fmt.Errorf("failed to open bundled attacher jar: %w", err)
+	}
+	defer j.closeFile(bundledAttacherFile)
+	tmpAttacherJarPath := tmpJavaAttacherJarPath(tempDir)
+	tmpAttacherJarFile, err := os.Create(tmpAttacherJarPath)
+	if err != nil {
+		defer j.deleteTempDir(tempDir)
+		return "", fmt.Errorf("failed to create tmp attacher jar: %w", err)
+	}
+	defer j.closeFile(tmpAttacherJarFile)
+	nBytes, err := io.Copy(tmpAttacherJarFile, bundledAttacherFile)
+	if err != nil {
+		defer j.deleteTempDir(tempDir)
+		return "", fmt.Errorf("failed to copy bundled attacher jar to %v: %w", bundledJavaAttacher, err)
+	}
+	j.logger.Debugf("%v (%v bytes) successfully copied to %v", bundledJavaAttacher, nBytes, tmpAttacherJarPath)
+	err = os.Chmod(tmpAttacherJarPath, 0755)
+	if err != nil {
+		j.logger.Errorf("failed to change permissions for %v to make it readable for all users: %v", tmpAttacherJarPath, err)
+	}
+	j.tempAttacherDir = tempDir
+	return tempDir, nil
+}
+
+func tmpJavaAttacherJarPath(tempDir string) string {
+	return filepath.Join(tempDir, javaAttacherJarName)
+}
+
+func (j *JavaAttacher) closeFile(file *os.File) {
+	err := file.Close()
+	if err != nil {
+		j.logger.Errorf("failed to close file %v: %v", file, err)
```

Comment:
The way I see it, I don't need to bother thinking about all possible errors and I assume they shouldn't occur, but if they do, for whatever reason I couldn't think of, I'd rather have it logged as it may indicate I am doing something not as I planned. Wherever I don't log something, it is because I am lazy 🙂 
But I definitely don't mind ignoring those and simplifying the code.

Reply:
Yeah I understand the rationale. I would sometimes do this too, but in this case I don't think it's justified. Failure to close the file means something more generally catastrophic has happened, and we'll find out soon enough from another avenue. Let's just do the defer for now.

Reply:
Oops, missed that, changing shortly